integr8ly/integreatly-operator
# MGDAPI-5462 Bump observability to v4.2.0

# Issue link
https://issues.redhat.com/browse/MGDAPI-5462

# What
Bumps the observability-operator to `v4.2.0`

# Verification steps
- Provision a cluster
- Checkout master and install RHOAM once the cluster is ready
```bash
export LOCAL=false
export USE_CLUSTER_STORAGE="true"
export INSTALLATION_TYPE=managed-api
make cluster/prepare/local
make code/run
```
- Wait until the install completes successfully
- Stop the operator locally
- Checkout this branch
- Re-run the operator
```bash
make code/rerun
```
- Wait until the upgrade has finished - version will be `4.2.0` and status will be `completed`
```bash
oc get rhmi rhoam -o redhat-rhoam-operator -o json | jq '.status.stages.installation.products.observability | .version, .status'
```
- Verify that the observability operator image is `quay.io/rhoas/observability-operator:v4.2.0`
```bash
oc get pods -l control-plane=controller-manager -n redhat-rhoam-observability-operator  -o json | jq '.items[0].spec.containers[0].image'
```
- Verify that the prometheus operator index image is `quay.io/integreatly/custom-prometheus-index:2.0.0`
```bash
oc get pods -l olm.catalogSource=prometheus-catalogsource -n redhat-rhoam-observability -o json | jq '.items[0].spec.containers[0].image'
```
- Verify that the oauth proxy image is `registry.redhat.io/openshift4/ose-oauth-proxy:v4.12`
```bash
oc get pods prometheus-prometheus-0 -n redhat-rhoam-observability -o json | jq '.spec.containers[] | select(.name == "oauth-proxy").image'
```
